dotcom-rendering/src/web/components/ImageComponent.tsx

```diff
@@ -265,10 +265,13 @@ export const ImageComponent = ({
 		getMaster(element.media.allImages) ??
 		getLargest(element.media.allImages);
 
-	const isSupported = (imageUrl: string) => {
+	const isSupported = (imageUrl: string): boolean => {
 		const supportedImages = ['jpg', 'jpeg', 'png'];
 		const extension = imageUrl.split('.').slice(-1)[0];
-		return extension && supportedImages.includes(extension.toLowerCase());
+		return (
+			extension !== undefined &&
+			supportedImages.includes(extension.toLowerCase())
+		);
 	};
```

Comment:
Maybe the logic could be simplified with the `endsWith` method?

```ts
const isSupported = (imageUrl: string): boolean => {
	const supportedImages = ['jpg', 'jpeg', 'png'];
	return supportedImages.some((extension) => imageUrl.endsWith(`.${extension}`));
};
```

Reply:
I was worried the creation of an array could be an issue, but it’s also 45% faster! 144 ns/iter v 213 ns/iter

```ts
const isSupportedSplit = (imageUrl: string): boolean => {
  const supportedImages = ["jpg", "jpeg", "png"];
  const extension = imageUrl.split(".").slice(-1)[0];
  return (
    extension !== undefined &&
    supportedImages.includes(extension.toLowerCase())
  );
};

const isSupportedEndsWith = (imageUrl: string): boolean => {
  const supportedImages = ["jpg", "jpeg", "png"];
  return supportedImages.some((extension) =>
    imageUrl.endsWith(`.${extension}`)
  );
};

Deno.bench("split", () => {
  const ok = isSupportedSplit("./some/image.jpg");
  const bad = isSupportedSplit("./some/image.svg");
  if (!ok || bad) throw Error("something went wrong");
});

Deno.bench("ends with", () => {
  const ok = isSupportedEndsWith("./some/image.jpg");
  const bad = isSupportedEndsWith("./some/image.svg");
  if (!ok || bad) throw Error("something went wrong");
});
```